joblib/test/test_hashing.py

```diff
@@ -403,6 +403,24 @@ def test_hashes_are_different_between_c_and_fortran_contiguous_arrays():
 
 
 @with_numpy
+def test_0d_array():
+    a = np.array(0)
+    b = np.array(0)
+    c = pickle.loads(pickle.dumps(b))
+    assert_equal(hash([a, b]), hash([a, c]))
```

Comment:
Why make the test more complicated than needed, just check that `hash(a) == hash(b)`.